src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,216 @@
+            addInside(value);
+        } else {
+            addIfLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        isPositionIndexBigger(index);
+        if (size == 0) {
+            add(value);
+            return;
+        }
+        if (index == 0 && size != 0) {
+            addAtFirstPosition(value);
+        } else if (index == size) {
+            addOnLastPositionWithIndex(value);
+        } else {
+            addInsideByIndex(index, value);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isPositionIndexBiggerOrEqualsSize(index);
+        return cycle(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isPositionIndexBiggerOrEqualsSize(index);
+        T oldValue = cycle(index).value;
+        cycle(index).value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        final Node<T> current = cycle(index);
+        if (size == 1) {
+            head = null;
+            tail = null;
+            size--;
+            return null;
+        }
+        Node<T> tempPrevious = cycle(index).previous;
+        Node<T> tempNext = cycle(index).next;
+        if (cycle(index).previous == null) {
+            tempNext.previous = null;
+            head = tempNext;
+            size--;
+            return current.value;
+        } else if (cycle(index).next == null) {
+            tempPrevious.next = null;
+            tail = tempPrevious;
+            size--;
+            return current.value;
+        } else {
+            tempPrevious.next = tempNext;
+            tempNext.previous = tempPrevious;
+            size--;
+            return current.value;
+        }
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        if (size == 1) {
+            head = null;
+            size--;
+            return true;
+        }
+        if (head.value.equals(object)) {
+            head = head.next;
+            size--;
+            return true;
+        }
+        Node<T> current = head;
+        int index = 0;
+        while (!Objects.equals(current.value, object)) {
+            current = current.next;
+            index++;
+            if (current == null) {
+                break;
+            }
+        }
+        if (index < size) {
+            remove(index);
+            return true;
+        } else {
+            return false;
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> cycle(int index) {
+        isPositionIndexBiggerOrEqualsSize(index);
```

Comment:
too long method name
```suggestion
        checkIndex(index);
```